src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -149,6 +164,47 @@ private UUID publishRequest(UUID consortiumId, PublicationRequest publicationReq
     return null;
   }
 
+  private void getSettingsForFailedTenants(UUID consortiumId, UUID publicationId, SharingSettingRequest sharingSettingRequest) {
+    boolean desiredResultReceived = false;
+    long startTime = System.currentTimeMillis();
+    long timeout = 3000;
+
+    while (Boolean.FALSE.equals(desiredResultReceived) && System.currentTimeMillis() - startTime < timeout) {
+      var publicationStatusEntity = publicationStatusRepository.findById(publicationId);
+      if (publicationStatusEntity.isPresent()) {
+        PublicationDetailsResponse publicationDetails = publicationService.getPublicationDetails(consortiumId, publicationId);
+        if (ObjectUtils.notEqual(publicationDetails.getStatus(), PublicationStatus.IN_PROGRESS)) {
+          desiredResultReceived = true;
+
+          Set<String> failedTenantList = publicationService
+            .getPublicationResults(consortiumId, publicationId)
+            .getPublicationResults()
+            .stream()
+            .filter(publicationResult -> HttpStatus.valueOf(publicationResult.getStatusCode()).isError())
+            .map(PublicationResult::getTenantId)
+            .collect(Collectors.toSet());
+
+          if (ObjectUtils.isNotEmpty(failedTenantList)) {
+            updateFailedSettingsToLocalSource(consortiumId, sharingSettingRequest, failedTenantList);
+          }
+        }
+      }
+    }
+  }
+
+  private void updateFailedSettingsToLocalSource(UUID consortiumId, SharingSettingRequest sharingSettingRequest, Set<String> failedTenantList) {
+    JsonNode payload = objectMapper.convertValue(sharingSettingRequest.getPayload(), JsonNode.class);
+    var updatedPayload = ((ObjectNode) payload).set(SOURCE, new TextNode(LOCAL_SETTING_SOURCE));
+
+    PublicationRequest publicationPutRequest = createPublicationRequestForSetting(sharingSettingRequest, HttpMethod.PUT.toString());
```

Comment:
Maybe I am wrong but we are sending a delete request so should HttpMethod be DELETE instead of PUT.

Reply:
Here we should use PUT, it is a case when DELETE operation was failed and and this case we do not want to remain setting with 'consortium' source, because it is not shared any more, so we need to produce PUT request to update source to 'local'